src/main/java/dev/blaauwendraad/masker/json/util/Utf8Util.java

```diff
@@ -31,6 +31,39 @@ public static int getCodePointByteLength(byte input) {
         throw new IllegalArgumentException("Input byte is not using UTF-8 encoding");
     }
 
+    /**
+     * Converts a 4-byte UTF-8 encoded character ('\u0000') into a char.
+     * Each byte MUST represent a valid HEX character, i.e.
+     * <ul>
+     *     <li>in range from {@code 48} ({@code '0'}) to {@code 57} ({@code '9'})
+     *     <li>in range from {@code 65} ({@code 'A'}) to {@code 70} ({@code 'F'})
+     *     <li>in range from {@code 97} ({@code 'a'}) to {@code 102} ({@code 'f'})
+     * </ul>
+     */
+    public static char unicodeHexToChar(byte b1, byte b2, byte b3, byte b4) {
+        int value = Character.digit(validateHex(b1), 16);
+        // since each byte transformed into a value, that is guaranteed to be in range 0 - 16 (4 bits)
+        // we shift by that amount
+        value = (value << 4) | Character.digit(validateHex(b2), 16);
+        value = (value << 4) | Character.digit(validateHex(b3), 16);
+        value = (value << 4) | Character.digit(validateHex(b4), 16);
+        return (char) value;
+    }
+
+    private static byte validateHex(byte hexByte) {
+        if (hexByte >= 48 && hexByte <= 57) {
+            return hexByte; // a digit from 0 to 9
+        }
+        if (hexByte >= 65 && hexByte <= 70) {
+            return hexByte; // a character from A to F
+        }
+        if (hexByte >= 97 && hexByte <= 102) {
+            return hexByte; // a character from a to f
+        }
+        throw new IllegalArgumentException("Invalid hex character '%s'".formatted((char) hexByte));
```

Comment:
Shouldn't this result in an (subclass of) InvalidJsonException (such as `InvalidJsonCharacterEncoding`) according to our contract?

Reply:
Eh? I guess it could be, but I don't think we have to go there with differentiating why exactly json is invalid. It's not like it would matter for the user in terms of handling the error

Reply:
Sure, so just regular Invalid Json Exception